### PR TITLE
Docs: Plugin migration - 4 intro docs

### PR DESCRIPTION
**Pull request by josmperez**
This PR migrates 4 files from grafana/docs/sources/developers/plugins into plugins-tools as part of the Dev Portal project. These files are added to the TOC under the "Introduction" label. 

Open question 1: When linking to Grafana as an external site, do we use /latest/?

Open question 2: Noting that folders in Hugo have landing pages but they apparently don't in Docusaurus (?), I think we should just not bring over the landing page content. Agree?

Open question 3: Noting that the "Getting Started" page is highlighted by deafault when a user clicks on "Docs" from the top of the website, do we want to change this to highlight "Backend plugins" instead? (And, how do we control this feature?)

Open question 4: Since Docusaurus apparently gives us the option of creating a TOC that is customized or one that is based automatically based on the oc structure, which approach is best suited for us? I'm not sure, but I'm leaning towards a customized TOC to give us maximum flexibility and avoid having very long file paths. However, the Hugo setup was based on the file system structure, so if we want to mirror Hugo for some reason then that's not the way to go. Is there a reason to mirror Hugo? 

Open question 5: I assume we need a robots no follow file on the site to avoid getting an SEO penalty. Not sure how/where to best add this?

### CODE REVIEW

**Comment by sympatheticmoose** [comment]
> Open question 1: When linking to Grafana as an external site, do we use /latest/?

Yes

> Open question 2: Noting that folders in Hugo have landing pages but they apparently don't in Docusaurus (?), I think we should just not bring over the landing page content. Agree?

Just to be sure - where landing pages are just ToC, agree. I think the only exception is https://grafana.com/docs/grafana/latest/developers/plugins/ but if there are others which have content then we'd need to catch them.

>Open question 3: Noting that the "Getting Started" page is highlighted by deafault when a user clicks on "Docs" from the top of the website, do we want to change this to highlight "Backend plugins" instead? (And, how do we control this feature?)

We can pick wherever makes the most sense, but Getting Started is fairly standard and would seem more suitable than backend plugins.

> Open question 4: Since Docusaurus apparently gives us the option of creating a TOC that is customized or one that is based automatically based on the oc structure, which approach is best suited for us? I'm not sure, but I'm leaning towards a customized TOC to give us maximum flexibility and avoid having very long file paths. However, the Hugo setup was based on the file system structure, so if we want to mirror Hugo for some reason then that's not the way to go. Is there a reason to mirror Hugo?

@jackw what led to the decision to go with custom? My _personal_ preference would be to use generated as then theres less to update. We can still impact the ordering (sidebar_position) and the display names (menu_title - i think).

> Open question 5: I assume we need a robots no follow file on the site to avoid getting an SEO penalty. Not sure how/where to best add this?

Probably makes sense to add temporarily whilst we're doing the migration, unless straightforward to add to the copied content only. This isn't an area I know anything about 🤷

File: docusaurus/docs/introduction/grafana-plugin-sdk-for-go.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+id: grafana-plugin-sdk-for-go
+title: Grafana plugin SDK for Go
+---
+
+# Grafana plugin SDK for Go
+
+The [Grafana plugin SDK for Go](https://pkg.go.dev/mod/github.com/grafana/grafana-plugin-sdk-go?tab=overview) is a [Go](https://golang.org/) module that provides a set of [packages](https://pkg.go.dev/mod/github.com/grafana/grafana-plugin-sdk-go?tab=packages) that you can use to implement a backend plugin.
+
+The plugin SDK provides a high-level framework with APIs, utilities, and tooling. By using the SDK, you can avoid the need to learn the details of the [plugin protocol]({{< relref "./plugin-protocol.md" >}}) and RPC communication protocol, so you don't have to manage either one.
```
Comment (jackw) [inline docusaurus/docs/introduction/grafana-plugin-sdk-for-go.md:10]:
```suggestion
The plugin SDK provides a high-level framework with APIs, utilities, and tooling. By using the SDK, you can avoid the need to learn the details of the [plugin protocol](./plugin-protocol.md) and RPC communication protocol, so you don't have to manage either one.
```

File: docusaurus/docs/introduction/plugin-protocol.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+From time to time, Grafana will offer additions of services, messages, and fields in the latest version of the plugin protocol. We don't expect these updates to introduce any breaking changes. However, if we must introduce breaking changes to the plugin protocol, we'll create a new major version of the plugin protocol.
+
+Grafana will release new major versions of the plugin protocol alongside new major Grafana releases. When this happens, we'll support both the old and the new plugin protocol for some time to make sure existing backend plugins continue to work.
+
+The plugin protocol attempts to follow Grafana's versioning. However, that doesn't mean we will automatically create a new major version of the plugin protocol when a new major release of Grafana is released.
+
+## Writing plugins without Go
+
+If you want to write a backend plugin in a language other than Go, then it's possible as long as the language supports gRPC. However, we recommend that you develop your plugin in Go for several reasons:
```
Comment (jackw) [inline docusaurus/docs/introduction/plugin-protocol.md:38]: I'm aware this PR is moving content across without edits but IMHO we probably shouldn't advertise this as an option for writing backend plugins in a language other than Go.

File: docusaurus/website/sidebars.js [hunk 3]
```diff
@@ -17,6 +17,11 @@ const sidebars = {
   // tutorialSidebar: [{ type: 'autogenerated', dirName: '.' }],
 
   docs: {
+    Introduction: [
+      'introduction/backend-plugins',
+      'introduction/grafana-plugin-sdk-for-go',
+      'introduction/plugin-protocol',
+      'introduction/data-frames',],
```
Comment (jackw) [inline docusaurus/website/sidebars.js:24]: Not sure if this fails CI but this file should be formatted via prettier:

```suggestion
      'introduction/data-frames',
    ],
```

**Comment by jackw** [comment]
>Open question 3: Noting that the "Getting Started" page is highlighted by deafault when a user clicks on "Docs" from the top of the website, do we want to change this to highlight "Backend plugins" instead? (And, how do we control this feature?)

> We can pick wherever makes the most sense, but Getting Started is fairly standard and would seem more suitable than backend plugins.

Agreed. Dropping users directly into backend plugins doesn't seem right to me as regardless of what they're building the getting started giving an intro to the tool and how to run it.

>Open question 2: Noting that folders in Hugo have landing pages but they apparently don't in Docusaurus (?), I think we should just not bring over the landing page content. Agree?

>Open question 4: Since Docusaurus apparently gives us the option of creating a TOC that is customized or one that is based automatically based on the oc structure, which approach is best suited for us?

> @jackw what led to the decision to go with custom? My personal preference would be to use generated as then theres less to update. We can still impact the ordering (sidebar_position) and the display names (menu_title - i think).

Custom wasn't a conscious decision, likely just evolved from the few hours I spent putting the site together. Switching to autogenerated should be fine but will likely need some tweaking in sidebar.js to get it working correctly. It should also allow for "landling pages" via naming files `index.md` or `README.md` e.g (`docs/myCategoryFolder/index.md`). I'm happy to sort out an autogenerated sidebar once this PR is merged.

File: docusaurus/docs/introduction/plugin-protocol.md (hunk 2 above)
Comment (josmperez) [reply docusaurus/docs/introduction/plugin-protocol.md:38]: That seems like a reasonable thing to do to me. @sympatheticmoose are you okay with removing this "Writing plugins without. Go" section?

**Comment by josmperez** [comment]
> Open question 3: Noting that the "Getting Started" page is highlighted by deafault when a user clicks on "Docs" from the top of the website, do we want to change this to highlight "Backend plugins" instead? (And, how do we control this feature?)
> 
> We can pick wherever makes the most sense, but Getting Started is fairly standard and would seem more suitable than backend plugins.
> 
> Agreed. Dropping users directly into backend plugins doesn't seem right to me as regardless of what they're building the getting started giving an intro to the tool and how to run it.

I think we should bear in mind that the content of [our current index page]([https://grafana.com/docs/grafana/latest/developers/plugins/] should probably become the new index page of the "Introduction" section, and then I see us making that the new highlighted page when users go to docs. In the Grafana doc content model, Introduction sections are at the top, followed by Get Started sections.

> Custom wasn't a conscious decision, likely just evolved from the few hours I spent putting the site together. Switching to autogenerated should be fine but will likely need some tweaking in sidebar.js to get it working correctly. It should also allow for "landling pages" via naming files index.md or README.md e.g (docs/myCategoryFolder/index.md). I'm happy to sort out an autogenerated sidebar once this PR is merged.

Agreed. I think either way could work for us, but if we go with autogenerated then I'd like to shorten the file names ad use the equivalent of menuTitle tags, so the URLs don't get quite so ugly.

File: docusaurus/docs/introduction/plugin-protocol.md (hunk 2 above)
Comment (josmperez) [reply docusaurus/docs/introduction/plugin-protocol.md:38]: removed section

File: docusaurus/docs/introduction/backend.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+## Grafana backend plugin system
+
+The Grafana backend plugin system is based on HashiCorp's [Go Plugin System over RPC](https://github.com/hashicorp/go-plugin). Our implementation of the Grafana server launches each backend plugin as a subprocess and communicates with it over [gRPC](https://grpc.io/).
+
+### Benefits for plugin development
+
+Grafana's approach has benefits for developers:
+
+- **Stability:** Plugins can't crash your Grafana process: a panic in a plugin doesn't panic the server.
+- **Ease of development:** Plugins can be written in any language that supports gRPC (for example, write a Go application and run `go build`).
```
Comment (sympatheticmoose) [inline docusaurus/docs/introduction/backend.md:37]: @jackw @josmperez This statement feels very close to https://github.com/grafana/plugin-tools/pull/296#discussion_r1257897345

How about we remove and for now have a generic statement like:

Ease of development: Grafana provides an officially supported SDK for Go and Tooling to help create plugins.

**Review by sympatheticmoose — APPROVED** [review]

File: docusaurus/docs/introduction/backend.md (hunk 4 above)
Comment (josmperez) [reply docusaurus/docs/introduction/backend.md]: done

**Review by jackw — APPROVED** [review]
LGTM! 🚀

**Pull request merged**